tests: address XFS 300M minimum size (#4361)

ThalesBarretto opened the pull request:
A [commit](<https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/commit/?id=6e0ed3d19c54603f0f7d628ea04b550151d8a262>) in xfsprogs raised the minimun size of a xfs filesystem to 300M, causing many tests to fail because they were trying to build even smaller filesystems.

This PR addresses the trivial cases by just resizing the file-based loopback devices and the relevant test parameters.

Fixes: #4361 
Reported-by: @mykaul
Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>

ThalesBarretto committed:
tests: shared-statfs.t XFS 300M minimum (#4361)

	A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/basic/posix/shared-statfs.t | 12 ++++++------
#  1 file changed, 6 insertions(+), 6 deletions(-)

ThalesBarretto committed:
tests: bug-1091935-brick-order-check-from-cli-to-glusterd.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  .../glusterd/bug-1091935-brick-order-check-from-cli-to-glusterd.t | 8 ++++----
#  1 file changed, 4 insertions(+), 4 deletions(-)

ThalesBarretto committed:
tests: df-results-post-replace-brick-operations.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/glusterd/df-results-post-replace-brick-operations.t | 10 +++++-----
#  1 file changed, 5 insertions(+), 5 deletions(-)

ThalesBarretto committed:
tests: bug-1586020-mark-dirty-for-entry-txn-on-quorum-failure.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  .../bug-1586020-mark-dirty-for-entry-txn-on-quorum-failure.t        | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)

ThalesBarretto committed:
tests: bug-948729-force.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/glusterd/bug-948729/bug-948729-force.t | 12 ++++++------
#  1 file changed, 6 insertions(+), 6 deletions(-)

ThalesBarretto committed:
tests: bug-948729.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/glusterd/bug-948729/bug-948729.t | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)

ThalesBarretto committed:
tests: distribute/file-create.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/basic/distribute/file-create.t | 10 +++++-----
#  1 file changed, 5 insertions(+), 5 deletions(-)

ThalesBarretto committed:
tests: bug-948729-mode-script.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/glusterd/bug-948729/bug-948729-mode-script.t | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

ThalesBarretto committed:
tests: arbiter-statfs.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/basic/afr/arbiter-statfs.t | 2 +-
#  1 file changed, 1 insertion(+), 1 deletion(-)
#  tests/basic/afr/arbiter-statfs.t | 2 +-
#  1 file changed, 1 insertion(+), 1 deletion(-)

ThalesBarretto committed:
tests: zero-fill-enospace.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/basic/posix/zero-fill-enospace.t | 6 ++++--
#  1 file changed, 4 insertions(+), 2 deletions(-)
#  tests/basic/posix/zero-fill-enospace.t | 6 ++++--
#  1 file changed, 4 insertions(+), 2 deletions(-)

ThalesBarretto committed:
tests: issue-1254-prioritize-enospc.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/replicate/issue-1254-prioritize-enospc.t | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)
#  tests/bugs/replicate/issue-1254-prioritize-enospc.t | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)

ThalesBarretto committed:
tests: fuse/bug-858488-min-free-disk.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/fuse/bug-858488-min-free-disk.t | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)
#  tests/bugs/fuse/bug-858488-min-free-disk.t | 6 +++---
#  1 file changed, 3 insertions(+), 3 deletions(-)

mykaul commented on tests/bugs/fuse/bug-858488-min-free-disk.t:
```diff
@@ -69,7 +70,7 @@ done
 ## Bring free space on one of the bricks to less than minfree value by
 ## creating one big file.
-dd if=/dev/zero of=$M0/fillonebrick.data bs=1024 count=25600 1>/dev/null 2>&1
+dd if=/dev/zero of=$M0/fillonebrick.data bs=1024 count=51200 1>/dev/null 2>&1
```
```suggestion
dd if=/dev/zero of=$M0/fillonebrick.data bs=4096 count=25600 1>/dev/null 2>&1
```

Time to get rid of this silly 1K.

ThalesBarretto replied:
Besides improving io alignment, it doubles the data written to the file (fom 50M to 100M). This still satisfies writing above the minfree=90% treshold on both bricks. But if we got that route there are many many other cases where "dd ... bs=1024"

mykaul commented:
We need to go over each test, to see that we don't know change the test meaning.

ThalesBarretto reviewed (commented):
Idk if this is the right place to change "bs=1024" to "bs=4096" just for the sake of io alignment. Maybe another patch set.

mykaul commented:
> Idk if this is the right place to change "bs=1024" to "bs=4096" just for the sake of io alignment. Maybe another patch set.

Well, I've tried in the past (https://github.com/gluster/glusterfs/pull/3860 ) . I think this was a good opportunity anyway, since you are changing the 'dd' command, to improve this.

ThalesBarretto commented:
> > Idk if this is the right place to change "bs=1024" to "bs=4096" just for the sake of io alignment. Maybe another patch set.
> 
> Well, I've tried in the past (#3860 ) . I think this was a good opportunity anyway, since you are changing the 'dd' command, to improve this.

Well, better to parametrize the writes then

ThalesBarretto committed:
tests:  tests/bugs/distribute/bug-1099890.t XFS 300M minimum (#4361)

        A commit in xfsprogs raised the minimun size of a xfs filesystem
to 300M, causing the tests to fail because they were trying to build
even smaller filesystems. This commit adjust the size parameters to fit.

Signed-off-by: Thales Antunes de Oliveira Barretto <thales.barretto.git@gmail.com>
#  tests/bugs/distribute/bug-1099890.t | 24 ++++++++++++------------
#  1 file changed, 12 insertions(+), 12 deletions(-)
#  tests/bugs/distribute/bug-1099890.t | 24 ++++++++++++------------
#  1 file changed, 12 insertions(+), 12 deletions(-)

ThalesBarretto committed:
Update tests/bugs/fuse/bug-858488-min-free-disk.t

Co-authored-by: Yaniv Kaul <yaniv.kaul@scylladb.com>#  tests/bugs/fuse/bug-858488-min-free-disk.t | 2 +-
#  1 file changed, 1 insertion(+), 1 deletion(-)
#  tests/bugs/fuse/bug-858488-min-free-disk.t | 2 +-
#  1 file changed, 1 insertion(+), 1 deletion(-)

pranithk commented on tests/basic/distribute/file-create.t:
```diff
@@ -23,8 +23,8 @@ TEST pidof glusterd
-TEST   truncate -s 25M $B0/brick1
-TEST   truncate -s 25M $B0/brick2
+TEST   truncate -s 300M $B0/brick1
+TEST   truncate -s 300M $B0/brick2
```
Line 64 in this test needs to fill the brick to more than 40% of the brick, this test fails on my local setup even after the change.
